### ISSUE TITLE
NET 7 RC2 version and SDK version mismatched when building on Ubuntu 22.04

### ISSUE THREAD

**Issue by mirespace** [dotnet/source-build#3078]
Hi team,

as discussed, and following https://github.com/dotnet/core/issues/7865#issuecomment-1281556370 where we find 7.0.100-rc.2.22517.1 versus tag 7.0.100-rc.2.22477.23, here is the complete output of the `dotnet --info` command and the link to the build log (https://launchpadlibrarian.net/629297430/buildlog_ubuntu-jammy-arm64.dotnet7_7.0.100-rc.2.22477.23~bootstrap+arm64-0ubuntu1~22.04.1_BUILDING.txt.gz)

```
.NET SDK:
 Version:   7.0.100-rc.2.22517.1
 Commit:    0a5360315a

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  22.04
 OS Platform: Linux
 RID:         ubuntu.22.04-arm64
 Base Path:   /usr/lib/dotnet/dotnet7-7.0.100-rc.2.22477.23/sdk/7.0.100-rc.2.22517.1/

Host:
  Version:      7.0.0-rc.2.22472.3
  Architecture: arm64
  Commit:       550605cc93

.NET SDKs installed:
  7.0.100-rc.2.22517.1 [/usr/lib/dotnet/dotnet7-7.0.100-rc.2.22477.23/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.0-rc.2.22476.2 [/usr/lib/dotnet/dotnet7-7.0.100-rc.2.22477.23/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.0-rc.2.22472.3 [/usr/lib/dotnet/dotnet7-7.0.100-rc.2.22477.23/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

The same also occurs for the x64 architecture  (I can also provide that log if needed.. or if anything else is needed!).

Thanks in advance!

Miriam

cc @omajid @crummel @MichaelSimons

**Comment by crummel** [comment]
Thanks for the logs!  I'm looking at this now.

**Comment by omajid** [comment]
This does seem to be a date leaking in.

I built the .NET 7 RC 2 last week and I got a `dotnet-sdk-7.0.100-rc.2.22511.1-rhel.8-x64.tar.gz` as the final SDK. I rebuilt the same sources today and got a `dotnet-sdk-7.0.100-rc.2.22517.1-rhel.8-x64.tar.gz`.

That's not great for reproducibility.

**Comment by omajid** [comment]
Hm, in RC 2, the generated `git-info/AllRepoVersions.props` contains `<installerOutputPackageVersion>7.0.100</installerOutputPackageVersion>` without any suffix.

**Comment by crummel** [comment]
I think the issue is that installer can't include itself in `eng/Version.Details.xml` so source-build can't get an OfficialBuildId that's stable for it.  I need to think a little bit about how we can actually solve this, but I think the issue should mostly go away once the installer version is stabilized.  There's still going to be a `Microsoft.DotNet.Sdk.Internal` package and a respective intermediate package produced with an unstabilized version but those artifacts shouldn't ship and shouldn't have user-visible version numbers.

**Comment by omajid** [comment]
Can we check if there's a git tag associated with current commit when building the source-tarbal? If there is, use that tag as the build id?

**Comment by MichaelSimons** [comment]
@crummel - please validate your asserting for 7.0.  For 8.0 please consider @omajid's suggestion.

**Comment by crummel** [comment]
I've verified this is fixed in the stabilized 7.0 builds.  I think the right approach going forward is to fix https://github.com/dotnet/source-build/issues/2250.  I've noted Omair's suggestion about the git tag there as well.
